src/pal/src/map/map.cpp

```diff
@@ -2241,7 +2241,9 @@ MAPmmapAndRecord(
     PAL_ERROR palError = NO_ERROR;
     LPVOID pvBaseAddress = NULL;
 
-    pvBaseAddress = mmap(addr, len, prot, flags, fd, offset);
+    off_t adjust = offset & (VIRTUAL_PAGE_MASK);
+
+    pvBaseAddress = mmap(static_cast<char *>(addr) - adjust, len + adjust, prot, flags, fd, offset - adjust);
```

Comment:
Ok, I've again missed that. I am sorry, I probably have an off day today. But in that case, it seems it would be good to set the PAL_MAX_PAGE_SIZE / SECTION_ALIGNMENT differently for 32 bit architectures (probably the same as we do for non-PAL in your change) to reduce the virtual address space consumption.